src/Development/IDE/LSP/Notifications.hs

```diff
@@ -69,4 +71,9 @@ setHandlersNotifications = PartialHandlers $ \WithMessage{..} x -> return x
             logInfo (ideLogger ide) $ "Files created or deleted: " <> msg
             modifyFileExists ide events
             setSomethingModified ide
+    ,LSP.didChangeWorkspaceFoldersNotificationHandler = withNotification (LSP.didChangeWorkspaceFoldersNotificationHandler x) $
+        \_ ide (DidChangeWorkspaceFoldersParams events) -> do
+            modifyWorkspaceFolders ide
+              $ HS.union      (foldMap (HS.singleton . parseWorkspaceFolder) (_added   events))
```

Comment:
```suggestion
              $ HS.union (HS.fromList (map parseWorkspaceFolder $ _added events)
```
I find that a bit easier to read and it avoids the (admittedly irrelevant) space leak of `foldMap` using a `foldr` instead of a `foldl'`.

Reply:
Wouldn't it be fun if ghcide would let us peek into the Core generated for an expression?